### DISCUSSION TITLE
How exactly does it infer minecraft version and mod version?

### DISCUSSION THREAD

**Discussion by qouteall** [Kira-NT/mc-publish#59] [Q&A] [answered]
Fabric's tutorial https://fabricmc.net/wiki/tutorial:publishing_mods_using_github_actions uses hardcoded version which I don't like. The README says that the mod version and game version can be inferred from the release tag. How exactly is it inferred? Should I use `1.2.3-mc1.19.4` or `mc1.19.4-1.2.3` or `v1.2.3-mc1.19.4`?

**Comment by Kira-NT** [comment, marked as answer]
In my personal opinion, I wouldn't rely on this resolution strategy, as it serves as a last resort solution, implemented for cases when no other information is available for the action to extract.

Since you are a mod developer, I assume that you are looking to publish a mod (either Fabric, Forge, or Quilt). In this case, you don't need to worry at all, because mc-publish will obtain the game version directly from the `minecraft` entry in the dependency list of your mod's metadata file (either `fabric.mod.json`, `mods.toml`, or `quilt.mod.json`, respectively). For example:

```json
"depends": {
  "minecraft": "1.18.x"
},
```

If, for some reason, you still want to use tags as a primary indicator of the Minecraft version in use, I would personally recommend adopting a Sodium-like naming convention, such as `mc1.19.4-1.2.3`. The action can certainly process that format.

Regarding the mod version, the release tag typically corresponds to the new version of your product, whatever it may be. As such, it will be used without any conversions performed on it (i.e., a `mc1.19.4-1.2.3` release tag will be propagated to Modrinth and/or CurseForge as the version "number" of your mod).

**Comment by qouteall** [comment]
Thanks for clarification
